### ISSUE TITLE
Error thrown after await in loop is swallowed

### ISSUE THREAD

**Issue by Arnavion** [MatAtBread/nodent#59]
http://nodent.mailed.me.uk/#function%20foo()%20%7B%0A%20%20%20%20throw%20new%20Error(%22foo%22)%3B%0A%7D%0A%0Aasync%20function%20bar()%20%7B%20%7D%0A%0Aasync%20function%20baz()%20%7B%0A%20%20%20%20do%20%7B%0A%20%20%20%20%20%20%20%20await%20bar()%3B%0A%20%20%20%20%7D%0A%20%20%20%20while(false)%3B%0A%0A%20%20%20%20%2F%2F%20throw%20new%20Error(%22foo%22)%3B%0A%20%20%20%20foo()%3B%0A%7D%0A%0Abaz().then(log%2C%20log)%3B%0A

```js
function foo() {
    throw new Error("foo");
}

async function bar() { }

async function baz() {
    do {
        await bar();
    }
    while(false);

    // throw new Error("foo");
    foo();
}

baz().then(log, log);
```

prints nothing.

If the normal `throw` line is uncommented, then the error is printed. I see that the codegen in latter case uses `$error` directly, which is why it works.

**Comment by matAtWork** [comment]
That is a very useful catch. I actually installed Nodent v3 on there today for testing (it's not released yet) - I've just flipped back to Nodent v2 (which is quite mature) and it works.

v3 does loops quite differently to avoid unecessary recursion, so you've provided an excellent test case for me which I'll add to the test suite. Thanks.

Meanwhile, if you're looking at something for production, stick to Nodent v2.
